DocumentsFromSnapshotMigration/src/test/java/com/rfs/EndToEndTest.java

```diff
@@ -0,0 +1,277 @@
+package com.rfs;
+
+import java.io.File;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import org.opensearch.migrations.metadata.tracing.MetadataMigrationTestContext;
+import org.opensearch.migrations.reindexer.tracing.DocumentMigrationTestContext;
+import org.opensearch.migrations.snapshot.creation.tracing.SnapshotTestContext;
+import org.opensearch.migrations.workcoordination.tracing.WorkCoordinationTestContext;
+
+import com.rfs.common.FileSystemRepo;
+import com.rfs.common.FileSystemSnapshotCreator;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.http.ConnectionContextTestParams;
+import com.rfs.framework.SearchClusterContainer;
+import com.rfs.http.ClusterOperations;
+import com.rfs.worker.SnapshotRunner;
+import lombok.Lombok;
+import lombok.SneakyThrows;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class EndToEndTest extends SourceTestBase {
+    @TempDir
+    private File localDirectory;
+
+    @ParameterizedTest(name = "Target OpenSearch {0}")
+    @ArgumentsSource(SupportedTargetCluster.class)
+    public void migrateFrom_ES_v6_8(final SearchClusterContainer.Version targetVersion) throws Exception {
+        var executorService = Executors.newFixedThreadPool(1);
```

Comment:
It would be simpler to remove the executor by having RFS run on the test thread.

Reply:
I added this because I needed to match the api of `migrateDocumentsSequentially`.  Not necessary now that I'll do `migrateDocumentsWithOneWorker` instead.  Will remove.